math/optimized_helper_functions.h

```diff
@@ -0,0 +1,100 @@
+/*
+	Copyright 2021 Kenn Sebesta
+
+	This file is part of the VESC firmware.
+
+	The VESC firmware is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The VESC firmware is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    */
+
+#ifndef OPTIMIZED_HELPER_FUNCTIONS_H_
+#define OPTIMIZED_HELPER_FUNCTIONS_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "datatypes.h"
+#include <math.h>
+#include <stdlib.h>
+
+// This takes advantage of the IEEE 754 standard, which requires that the sign
+// bit of single-precision floats is the last bit.
+// It is faster than `signbit()` because it does not have a bit shift and does no type checking.
+// On the flip side, it is dangerous because it does no checking of any kind. It will
+// not care if the input is not a `float`, if it is `NaN`, etc...
+// It is guaranteed to work on all systems which adhere to IEEE 754.
+// This works on both big- and little-endian systems.
+// The macro returns 1 if a `float` is negative, 0 if otherwise
```

Comment:
Actually it returns 1<<31 or 0

Reply:
Excellent point, I'll fix that before taking this PR out of draft mode.